federation/ipatuura/README.md

```diff
@@ -0,0 +1,221 @@
+```
+
+You can also test the AD integration is working in the bridge using `kinit`:
+
+```
+$ kinit Administrator@KEYCLOAK.ORG
+Password for Administrator@KEYCLOAK.ORG:
+Warning: Your password will expire in 36 days on Wed Jan  8 15:24:44 2025
+```
+
+```
+$ klist
+Ticket cache: KEYRING:persistent:0:0
+Default principal: Administrator@KEYCLOAK.ORG
+
+Valid starting     Expires            Service principal
+12/02/24 20:01:56  12/03/24 06:01:56  krbtgt/KEYCLOAK.ORG@KEYCLOAK.ORG
+renew until 12/09/24 20:01:54
+```
+
+This ensures your `krb5.conf` is working.
+
+Now leave the session open and listening to the `httpd` error logs to catch error messages in case you face errors when adding the integration domain in the next step:
+
+```
+$ tail -f /var/log/httpd/error_log
+```
+
+## Add an integration domain to the bridge service
+
+The integration domain is configured in a JSON file that has to be sent to the bridge via HTTP. Send the following request from the actual host:
+
+```
+$ curl -v -k -X POST "https://ipa-bridge.keycloak.org/domains/v1/domain/" -H "accept: application/json" -H "Content-Type: application/json" -H "X-CSRFToken: x1yU9RGPKs4mJdWIOzEc7wKbwbnJ0B6iTHuW6ja0gdBpEOBVacK1vIhSSYlfsnRw" -d @integrationdomain.json
+```
+
+The `integrationdomain.json` file looks like this:
+
+```
+{
+   "id": "1",
+   "name": "keycloak.org",
+   "description": "AD Integration Domain",
+   "integration_domain_url": "ldap://DC1.keycloak.org",
+   "client_id": "Administrator@KEYCLOAK.ORG",
+   "client_secret": "Passw0rd",
+   "id_provider": "ad",
+   "user_extra_attrs": "mail:mail, sn:sn, givenname:givenname",
+   "user_object_classes": "user,organizationalPerson,person,top",
+   "users_dn": "CN=Keycloak,CN=Users,DC=keycloak,DC=org",
+   "ldap_tls_cacert": "/etc/openldap/certs/cacert.pem",
+   "keycloak_hostname": "keycloak.ipa.test"
+}
+```
+
+### Important observations:
+1- `"name"` must match the AD domain name (i.e. `keycloak.org`)
+
+2- in `"integration_domain_url"`, use `ldap` instead of `ldaps`
+
+3- the `"ldap_tls_cert"` is not actually used in this setup, just set any value
+
+4- `"keycloak_hostname"` must resolve to the host running keycloak (see step above to ensure it was added to `/etc/hosts`)
+
+## Setup the IPA-Tuura user federation in keycloak
+
+First start keycloak with the `ipa-tuura-federation` feature enabled:
+
+```
+$ ./kc.sh start-dev --debug --db=postgres --db-username=<youruser> --db-password=<yourpass> --features=ipa-tuura-federation
```

Comment:
```suggestion
$ ./kc.sh start-dev --features=ipa-tuura-federation
```

Reply:
The suggestion above is to make it more clear that the only thing you need is the feature enabled.

Reply:
Agree, will send update!

Reply:
Done!